index.js

```diff
@@ -3232,6 +3232,51 @@
   S.range =
   def('range', {}, [$.Integer, $.Integer, $.Array($.Integer)], range);
 
+  //# groupBy :: (a -> a -> Boolean) -> Array a -> Array (Array a)
+  //.
+  //. Splits its list argument into a list of lists of equal,
+  //. adjacent elements. The equality of the elements is tested with
+  //. a curried function provided by the user in the first argument.
+  //. The behavior of groupBy aims to be equal to Haskell's groupby,
+  //. rather than Ramda's groupBy.
+  //.
+  //. The behavior of groupBy can be surprising for functions that
+  //. aren't reflexive, transitive and symmetric
+  //. (see [equivalence][] relation).
+  //.
+  //. ```javascript
+  //. > S.groupBy(S.equals, [1, 1, 2, 3, 3, 4])
+  //. [[1, 1], [2], [3, 3], [4]]
```

Comment:
It would be nice to have an example that shows what happens if an equal value reoccurs after an unequal one.